ADAL/src/cache/ADTokenCacheAccessor.m

```diff
@@ -342,52 +359,52 @@ - (void)removeItemFromCache:(ADTokenCacheItem *)cacheItem
         return;
     }
     
-    NSUUID* correlationId = [context correlationId];
+    
+    ADTelemetryCacheEvent* event = [[ADTelemetryCacheEvent alloc] initWithName:AD_TELEMETRY_EVENT_TOKEN_CACHE_DELETE
+                                                                       context:context];
+    [event setSpeInfo:cacheItem.speInfo];
     [[ADTelemetry sharedInstance] startEvent:[context telemetryRequestId] eventName:AD_TELEMETRY_EVENT_TOKEN_CACHE_DELETE];
-    BOOL removed = NO;
+    [self removeImpl:cacheItem refreshToken:refreshToken context:context error:error];
+    [[ADTelemetry sharedInstance] stopEvent:[context telemetryRequestId] event:event];
+}
+
+- (void)removeImpl:(ADTokenCacheItem *)cacheItem
+      refreshToken:(NSString *)refreshToken
+           context:(id<ADRequestContext>)context
+             error:(ADAuthenticationError *)error
+{
     //The refresh token didn't work. We need to tombstone this refresh item in the cache.
-    ADTokenCacheKey* exactKey = [cacheItem extractKey:nil];
-    if (exactKey)
+    ADTokenCacheKey* cacheKey = [cacheItem extractKey:nil];
+    if (!cacheKey)
     {
-        ADTokenCacheItem* existing = [_dataSource getItemWithKey:exactKey userId:cacheItem.userInformation.userId correlationId:correlationId error:nil];
-        if ([refreshToken isEqualToString:existing.refreshToken])//If still there, attempt to remove
-        {
-            AD_LOG_VERBOSE_F(@"Token cache store", correlationId, @"Tombstoning cache for resource: %@", cacheItem.resource);
-            //update tombstone property before update the tombstone in cache
-            [existing makeTombstone:@{ @"correlationId" : [correlationId UUIDString],
-                                       @"errorDetails" : [error errorDetails],
-                                       @"protocolCode" : [error protocolCode] }];
-            [_dataSource addOrUpdateItem:existing correlationId:correlationId error:nil];
-            removed = YES;
-        }
+        return;
     }
     
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the useless trailing whitespaces at the end of this line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=objc%3AS1131)